File: src/main/java/com/dauphine/blogger/services/Post.java
package com.dauphine.blogger.services;
import java.time.LocalDateTime;

import com.dauphine.blogger.models.Category;
import java.util.UUID;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
@Entity
@Table(name = "post")
public class Post {

    @Id
    @Column(name = "id")
    private UUID id;

    @Column(name = "title")
    private String title;

    @Column(name = "content")
    private String content;

    @Column(name = "created_date")
    private LocalDateTime createdDate;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    public Post() {
    }

    public Post(String title, String content, Category category) {
        this.id = UUID.randomUUID();
        this.title = title;
        this.content = content;
        this.createdDate = LocalDateTime.now();
        this.category = category;
    }
}

File: src/app/data/post.ts
import {Category} from "./category";
import { FormsModule,ReactiveFormsModule } from '@angular/forms';

export interface Post {
    id: string;
    title:string;
    content: string;
    createdDate: string;
    category: Category;
    
}

export type PostWithoutID = {
  title: string;
  category: { id: string; name: string };  // Ensure category is an object
  content: string;
};

export type PostWithTimestamp = Post & { timestamp: string };